CONTRIBUTING.md

```diff
@@ -27,3 +26,25 @@ Specifically:
 - We don't merge code into `main` that isn't releasable
 - We perform automated testing on all changes before they get merged to `main`
 - We create immutable release artifacts
+
+## How to Contribute
+Please ensure there is a Gitub issue for your proposed change, this helps the UDS CLI team to understand the context of the change and to track the progress of the work. If there isn't an issue for your change, please create one before starting work. The recommended workflow for contributing is as follows:
+
+1. **Fork this repo** and clone it locally
+2. **Create a branch** for your changes
+3. **Create and [test](#testing)** your changes
+4. **Push your branch** to your fork
+5. **Open a PR** against the `main` branch of this repo
+
+### Testing
+
+We strive to test all changes made to UDS CLI. If you're adding a new feature or fixing a bug, please add tests to cover the new functionality. Unit tests and E2E tests are both welcome, but we leave it up to the contributor to decide which is most appropriate for the change. Below are some guidelines for testing:
```

Comment:
Again probably not 100% necessary in this PR but would be nice to have a quick "how to run tests" so that someone can validate the tests they created + existing ones against their changes.

Reply:
added!